### PR TITLE
Implement NetworkNamespace resource

### PR DESCRIPTION
**Pull request by ysksuzuki**
This PR implements NetworkNamespace resource.

- Create Linux network namespaces as specified with a yaml representation.
- Create veth pairs and add the host-side interface to the bridge specified.
- Run init scripts and commands inside the network namespace as specified.

### CODE REVIEW

**Review by ymmt2005 — CHANGES_REQUESTED** [review]
Please fix as suggested.

**Comment by ysksuzuki** [comment]
Requested changes are

- Move Spec types which represents specification in Yaml to types package
- Implement validate function in types package
- Rename netns.Start function to netns.Setup
- Make NetworkNamespace's apps field optional
- Make NetworkNamespace's interfaces field required

File: v2/pkg/types/cluster.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+	}
+	return cluster, nil
+}
+
+func readSingleYamlDoc(reader io.Reader) (string, error) {
+	buf := make([]byte, 1024)
+	maxBytes := 16 * 1024 * 1024
+	base := 0
+	for {
+		n, err := reader.Read(buf[base:])
```
Comment (ymmt2005) [inline v2/pkg/types/cluster.go:157]: Why don't you use `ioutil.ReadAll`?
https://golang.org/pkg/io/ioutil/#ReadAll

File: v2/pkg/types/cluster.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+				buf = append(buf, make([]byte, len(buf))...)
+				continue
+			}
+			return "", errors.New("yaml document is too large")
+		}
+		if err != nil {
+			return "", err
+		}
+		base += n
+		return string(buf[:base]), nil
```
Comment (ymmt2005) [inline v2/pkg/types/cluster.go:173]: I think it is more convenient and reasonable to return a byte slice rather than a string.

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

File: v2/pkg/types/cluster.go (hunk 1 above)
Comment (ysksuzuki) [reply v2/pkg/types/cluster.go:157]: I want to read each resource one by one from a yaml document, but `ioutil.ReadAll` reads all bytes until EOF at once. So it seems not to work for me.

Comment (ymmt2005) [reply v2/pkg/types/cluster.go:157]: That's done by `json.YAMLFramer.NewFrameReader`.

Comment (ysksuzuki) [reply v2/pkg/types/cluster.go:157]: I tried it like following, but that's not what I expected. Did I miss something?

```
	yaml := `
kind: Network
name: internet
type: external
use-nat: true
---
kind: Network
name: core-to-op
type: internal
use-nat: false

`
	f := json.YAMLFramer.NewFrameReader(ioutil.NopCloser(strings.NewReader(yaml)))
        // Returns all bytes contains two network resources
	bytes, err := ioutil.ReadAll(f)
```

Comment (ymmt2005) [reply v2/pkg/types/cluster.go:157]: Ah, I'm sorry, I was wrong.

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**